components/widgets/numericeditbox.cpp

```diff
@@ -30,15 +28,15 @@ namespace Gui
 
         try
         {
-            mValue = boost::lexical_cast<int>(newCaption);
+            mValue = std::stoi(newCaption);
             int capped = std::min(mMaxValue, std::max(mValue, mMinValue));
             if (capped != mValue)
             {
                 mValue = capped;
                 setCaption(MyGUI::utility::toString(mValue));
             }
         }
-        catch (boost::bad_lexical_cast&)
+        catch (...)
```

Comment:
Both of these, I guess?
<pre>
If no conversion could be performed, an invalid_argument exception is thrown.

If the value read is out of the range of representable values by an int, an out_of_range exception is thrown.
</pre>